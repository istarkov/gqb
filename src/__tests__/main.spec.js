import { gqb, getResolvers } from '../';
import { graphql } from 'graphql'; // eslint-disable-line
import { makeExecutableSchema } from 'graphql-tools'; // eslint-disable-line

test('it works', async () => {
  const schema = gqb`
    type Query {
      hello: String${
        () => 'World'
      },
      pic(size: Int): String${
        (_, { size }) => `pic size is: ${size}`
      }
    }

    type Mutation {
      setMessage(message: String): String${
        (_, { message }) => `Mutated ${message}`
      }
    }
  `;

  const executableSchema = makeExecutableSchema(
    getResolvers(schema)
  );

  const { data } = await graphql(
    executableSchema,
    '{ hello, pic(size: 20)}'
  );

  expect(data).toMatchSnapshot();

  const { data: mutData } = await graphql(
    executableSchema,
    'mutation { setMessage(message: "Hello") }'
  );

  expect(mutData).toMatchSnapshot();
});
