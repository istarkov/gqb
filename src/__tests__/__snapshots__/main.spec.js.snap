exports[`test it works 1`] = `
Object {
  "hello": "World",
  "pic": "pic size is: 20",
}
`;

exports[`test it works 2`] = `
Object {
  "setMessage": "Mutated Hello",
}
`;
